packages/arb-token-bridge-ui/src/util/withdrawals/helpers.ts

```diff
@@ -58,19 +67,73 @@ export async function mapETHWithdrawalToL2ToL1EventResult(
   }
 }
 
+function getFirstExecutableBlockFromCache({
+  event,
+  l2ChainID
+}: {
+  event: L2ToL1EventResult
+  l2ChainID: number
+}) {
+  const cacheKey = getMessagesCacheKey({ event, l2ChainId: l2ChainID })
+  const firstExecutableBlockCache = JSON.parse(
+    localStorage.getItem(firstExecutableBlockLocalStorageKey) || '{}'
+  )
+  const result = firstExecutableBlockCache[cacheKey]
+
+  if (!result) {
+    return undefined
+  }
+  return Number(result)
+}
+
+export function saveFirstExecutableBlockToCache(key: string, value: number) {
+  const currentCache = JSON.parse(
+    localStorage.getItem(firstExecutableBlockLocalStorageKey) || '{}'
+  )
+  localStorage.setItem(
+    firstExecutableBlockLocalStorageKey,
+    JSON.stringify({ ...currentCache, [key]: value })
+  )
+}
+
+function saveExecutedMessageToCache(key: string) {
+  const currentCache = JSON.parse(
+    localStorage.getItem(executedMessagesLocalStorageKey) || '{}'
+  )
+  localStorage.setItem(
+    executedMessagesLocalStorageKey,
+    JSON.stringify({ ...currentCache, [key]: true })
+  )
+}
+
 export async function getOutgoingMessageState(
   event: L2ToL1EventResult,
   l1Provider: Provider,
   l2Provider: Provider,
-  l2ChainID: number
+  l2ChainID: number,
+  currentParentChainBlock: number
 ) {
-  const cacheKey = getExecutedMessagesCacheKey({
```

Comment:
We should do the check for executed messages cache first, because if it's there, then we don't need to do anything else

Reply:
done